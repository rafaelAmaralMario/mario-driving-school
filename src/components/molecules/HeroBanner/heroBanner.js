import React from "react";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";

import { Container } from 'components/atoms'

const HeroContainer = styled.div`
  color: #fff;
  padding: 100px;
  ${(props) => css`
    background: url(${props.image}), rgba(0, 0, 0, 0.4);
    background-size: cover;
    background-position: center;
    background-blend-mode: overlay;
  `}
`;

const Content = styled.div`
  display: inline-block;
  p,
  li {
    font-size: 20px;
    font-weight: 300;
  }
  ul {
    list-style: none;
    padding-left: 0;
  }
  li::before {
    color: ${ props => props.theme.colors.primary.main};
    content: "\\2713\\0020";
  }
`;

const HeroBanner = ({ image, children }) => (
  <HeroContainer image={image} data-testid="hero-container">
    <Container>
      <Content>{children}</Content>
    </Container>
  </HeroContainer>
);

HeroBanner.propTypes = {
  /**
   * Hero Banner Background Image
   */
  image: PropTypes.string,
  /**
   * Hero Banner Elements
   */
  children: PropTypes.node,
};

HeroBanner.defaultProps = {
  title: "Hero Banner Default Title",
};

export default HeroBanner;
