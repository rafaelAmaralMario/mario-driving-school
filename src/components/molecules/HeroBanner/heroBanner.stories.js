import React from "react";
import { text } from "@storybook/addon-knobs";

import HeroBanner from "./heroBanner";
import Heading from 'components/atoms/Heading/Heading';
import Button from 'components/atoms/Button/Button';
import BG1 from "stories/assets/bg-1.jpg";
import BG2 from "stories/assets/bg-2.jpg";

export default {
  title: "Components|Molecules/HeroBanner",
  component: HeroBanner,
};

export const withParagraph = () => (
  <HeroBanner image={BG1}>
    <Heading><h1>{text("Title", "Earn Your freedom to go anywhere.")}</h1></Heading>
    <p>{text("Paragraph", "Lorem ipsum dolor sit amet consectetur adipisicing elit.")}</p>
  </HeroBanner>
);

export const withList = () => (
  <HeroBanner image={BG2}>
    <Heading><h1>Earn Your <strong>freedom</strong><br /> to go anywhere.</h1></Heading>
    <ul className="hero-banner">
      <li className="hero-banner-list-item">
        Lorem ipsum dolor sit amet consectetur adipisicing elit.
      </li>
      <li className="hero-banner-list-item">
        Aperiam odio, eum beatae dignissimos eveniet excepturi?
      </li>
      <li className="hero-banner-list-item">
        Porro quibusdam labore beatae veritatis optio debitis, corrupti ea aut
        nam voluptas qui incidunt officiis.
      </li>
    </ul>
    <Button variant="outlined" color="primary">Apply Now</Button>
  </HeroBanner>
);
