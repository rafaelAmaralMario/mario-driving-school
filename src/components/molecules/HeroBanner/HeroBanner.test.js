import React from 'react'
import { render } from 'utils/test-utils';

import HeroBanner from './heroBanner';

test('render herobanner with children ', () => {
  const { getByText } = render(
    <HeroBanner image="">Mario Driving School</HeroBanner>
  )

  expect(getByText("Mario Driving School")).toBeInTheDocument();
})

test("should render with an image", () => {
  const image = "http://test.com/image.jpg"
  const { asFragment } = render(
    <HeroBanner image={image}>Mario Driving School</HeroBanner>
  );

  expect(asFragment()).toMatchSnapshot();
});

test('should render herobanner with children and Image ', () => {
  const image = "http://test.com/image.jpg"

  const { getByTestId } = render(
    <HeroBanner image={image}>Mario Driving School</HeroBanner>
  )

  expect(getByTestId("hero-container")).toHaveStyleRule({
    backgroundImage: image
  });
})


