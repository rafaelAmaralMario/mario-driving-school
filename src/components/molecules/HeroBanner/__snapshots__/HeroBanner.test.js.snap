// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with an image 1`] = `
<DocumentFragment>
  .c1 {
  width: 100%;
  padding: 0 8px;
}

.c0 {
  color: #fff;
  padding: 100px;
  background: url(http://test.com/image.jpg),rgba(0,0,0,0.4);
  background-size: cover;
  background-position: center;
  background-blend-mode: overlay;
}

.c2 {
  display: inline-block;
}

.c2 p,
.c2 li {
  font-size: 20px;
  font-weight: 300;
}

.c2 ul {
  list-style: none;
  padding-left: 0;
}

.c2 li::before {
  color: #ffc107;
  content: "\\2713\\0020";
}

@media (min-width:undefinedpx) {
  .c1 {
    padding: 0 16px;
  }
}

@media (min-width:992px) {
  .c1 {
    max-width: 1140px;
    margin: 0 auto;
    padding: 0;
  }
}

<div
    class="c0"
    data-testid="hero-container"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Mario Driving School
      </div>
    </div>
  </div>
</DocumentFragment>
`;
